hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect } from 'react';
+
+const mediaQueryChangeSubscribe = (mq: MediaQueryList, handler: () => void) => {
+  if (mq.addEventListener) {
+    mq.addEventListener('change', handler);
+  } else {
+    mq.addListener(handler);
+  }
+};
+
+const mediaQueryChangeUnsubscribe = (
+  mq: MediaQueryList,
+  handler: () => void
+) => {
+  if (mq.removeEventListener) {
+    mq.removeEventListener('change', handler);
+  } else {
+    mq.removeListener(handler);
+  }
+};
+
+export function useMediaQuery(query: string): boolean | undefined {
+  const [matches, setMatches] = useState<boolean>();
+
+  useEffect(() => {
+    if (typeof window.matchMedia === 'function') {
```

Comment:
```suggestion
    if (typeof window?.matchMedia === 'function') {
```

Reply:
window might not be defined. In theory it should as effects are always client-side, but just to be in the safe side of things!

Reply:
Got your point and fixed